exercises/robot-name/example.scala

```diff
@@ -1,21 +1,18 @@
 import scala.util.Random
 
-class Robot {
-  private var _name: String = makeName
-
-  def name: String = _name
-
-  def reset(): Unit = _name = makeName
-
-  private def makeName = prefix + suffix
-
-  private def prefix = Random.shuffle(Robot.alphabet).take(2).mkString
-
-  private def suffix = Random.shuffle(Robot.suffixes).head.toString
+object UniqueNames {
+  val names =
+    Random.shuffle(1 to 26 * 26 * 1000).iterator.map(n => {
+      val lettersPart = n / 1000
```

Comment:
Does this really generate all possible names, or aren't some `n`s mapped to the same name?
But well, if your new test succeeds then that's the answer ... :)